File: C/TutorialLevel2Bot/GameCommander.cpp
#include "GameCommander.h"

using namespace MyBot;

GameCommander::GameCommander()
{
}

void GameCommander::onStart() 
{
}

void GameCommander::onEnd(bool isWinner)
{
}

void GameCommander::onFrame()
{
	// 아군 베이스 위치. 적군 베이스 위치 정보를 저장/업데이트한다
	InformationManager::Instance().update();
		
	// 플레이어 정보 표시 - InformationManager 의 멤버변수 사용
	BWAPI::Broodwar->drawTextScreen(5, 5, "My Player: %c%s (%s) ",
		BWAPI::Broodwar->self()->getTextColor(), BWAPI::Broodwar->self()->getName().c_str(), InformationManager::Instance().selfRace.c_str());
	BWAPI::Broodwar->drawTextScreen(5, 15, "Enemy Player: %c%s (%s)",
		BWAPI::Broodwar->enemy()->getTextColor(), BWAPI::Broodwar->enemy()->getName().c_str(), InformationManager::Instance().enemyRace.c_str());

	// 현재 FrameCount 표시
	BWAPI::Broodwar->drawTextScreen(300, 100, "FrameCount: %d", BWAPI::Broodwar->getFrameCount());

	// 유닛 id 표시
	for (auto & unit : BWAPI::Broodwar->getAllUnits()) {
		BWAPI::Broodwar->drawTextMap(unit->getPosition().x, unit->getPosition().y, "%d", unit->getID());
	}

	// 플레이어 Start Location 표시 - InformationManager 의 멤버변수 사용
	if (InformationManager::Instance()._mainBaseLocations[BWAPI::Broodwar->self()]) {
		BWAPI::Broodwar->drawTextScreen(200, 5, "Start Location: %d, %d",
			InformationManager::Instance()._mainBaseLocations[BWAPI::Broodwar->self()]->getTilePosition().x,
			InformationManager::Instance()._mainBaseLocations[BWAPI::Broodwar->self()]->getTilePosition().y);
	}
	if (InformationManager::Instance()._mainBaseLocations[BWAPI::Broodwar->enemy()]) {
		BWAPI::Broodwar->drawTextScreen(200, 15, "Start Location: %d, %d",
			InformationManager::Instance()._mainBaseLocations[BWAPI::Broodwar->enemy()]->getTilePosition().x,
			InformationManager::Instance()._mainBaseLocations[BWAPI::Broodwar->enemy()]->getTilePosition().y);
	}
}

void GameCommander::onUnitShow(BWAPI::Unit unit)			
{ 
}

void GameCommander::onUnitHide(BWAPI::Unit unit)			
{
}

void GameCommander::onUnitCreate(BWAPI::Unit unit)		
{ 
}

void GameCommander::onUnitComplete(BWAPI::Unit unit)
{
}

void GameCommander::onUnitDestroy(BWAPI::Unit unit)		
{
}

void GameCommander::onUnitRenegade(BWAPI::Unit unit)
{
}

void GameCommander::onUnitMorph(BWAPI::Unit unit)
{ 
}

void GameCommander::onUnitDiscover(BWAPI::Unit unit)
{
}

void GameCommander::onUnitEvade(BWAPI::Unit unit)
{
}

File: JAVA/TutorialLevel2Bot/src/GameCommander.java
import bwapi.Unit;

public class GameCommander {
	
	public void onStart() 
	{
	}

	public void onEnd(boolean isWinner)
	{
	}

	public void onFrame()
	{
		// 아군 베이스 위치. 적군 베이스 위치 정보를 저장/업데이트한다
		InformationManager.Instance().update();

		// 플레이어 정보 표시 - InformationManager 의 멤버변수 사용
		MyBotModule.Broodwar.drawTextScreen(5, 5, "My Player: "+MyBotModule.Broodwar.self().getTextColor()+MyBotModule.Broodwar.self().getName()
				+" ("+InformationManager.Instance().selfRace+")");
		MyBotModule.Broodwar.drawTextScreen(5, 15, "Enemy Player: "+MyBotModule.Broodwar.enemy().getTextColor()
				+MyBotModule.Broodwar.enemy().getName()+" ("+InformationManager.Instance().enemyRace+")");
		
		// 현재 FrameCount 표시
		MyBotModule.Broodwar.drawTextScreen(300, 100, "FrameCount: "+MyBotModule.Broodwar.getFrameCount());

		// 유닛 id 표시
		for (Unit unit : MyBotModule.Broodwar.getAllUnits()) {
			MyBotModule.Broodwar.drawTextMap(unit.getPosition().getX(), unit.getPosition().getY(), ""+unit.getID());
		}		
		
		// 플레이어 Start Location 표시 - InformationManager 의 멤버변수 사용
		if (InformationManager.Instance().mainBaseLocations.get(MyBotModule.Broodwar.self()) != null) {
			MyBotModule.Broodwar.drawTextScreen(200, 5, "Start Location: "
				+InformationManager.Instance().mainBaseLocations.get(MyBotModule.Broodwar.self()).getTilePosition().getX()+","
				+InformationManager.Instance().mainBaseLocations.get(MyBotModule.Broodwar.self()).getTilePosition().getY());
		}
		if (InformationManager.Instance().mainBaseLocations.get(MyBotModule.Broodwar.enemy()) != null) {
			MyBotModule.Broodwar.drawTextScreen(200, 15, "Start Location: "
				+InformationManager.Instance().mainBaseLocations.get(MyBotModule.Broodwar.enemy()).getTilePosition().getX()+","
				+InformationManager.Instance().mainBaseLocations.get(MyBotModule.Broodwar.enemy()).getTilePosition().getY());
		}
	}

	public void onUnitShow(Unit unit) { 
	}

	public void onUnitHide(Unit unit) {
	}

	public void onUnitCreate(Unit unit) { 
	}

	public void onUnitComplete(Unit unit) {
	}

	public void onUnitDestroy(Unit unit) {
	}

	public void onUnitRenegade(Unit unit) {
	}

	public void onUnitMorph(Unit unit) {
	}

	public void onUnitDiscover(Unit unit) {		
	}

	public void onUnitEvade(Unit unit) {		
	}
}